From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Tue, 9 Dec 2025 17:19:52 +0000
Subject: arm64: dts: sun50i-h616-x96-mate: Enable EMAC1 with RMII PHY

Enable EMAC1 ethernet controller with external RMII PHY for the
X96 Mate TV box.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
Signed-off-by: Werner <werner@armbian.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
index 085f3e4e8eaa..20744b349569 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-x96-mate.dts
@@ -16,6 +16,7 @@ / {
 	compatible = "hechuang,x96-mate", "allwinner,sun50i-h616";

 	aliases {
+		ethernet0 = &emac1;
 		serial0 = &uart0;
 	};

@@ -55,6 +56,24 @@ &gpu {
 	status = "okay";
 };

+&emac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins>;
+	phy-mode = "rmii";
+	phy-handle = <&rmii_phy>;
+	phy-supply = <&reg_aldo1>;
+	allwinner,rx-delay-ps = <3100>;
+	allwinner,tx-delay-ps = <700>;
+	status = "okay";
+};
+
+&mdio1 {
+	rmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
 &ir {
 	status = "okay";
 };
--
Armbian
